### DISCUSSION TITLE
Contribute public servers

### DISCUSSION THREAD

**Discussion by 0x100001** [rustdesk/rustdesk#1657] [General]
Hello folks,

since I discovered rustdesk, I am using it on a regualy basis and I love it.
To give back, I would like to host a german and america public server for the community.
Is that possible and if so, what are the guidelines? 
I am owning a cyber security company. Before posting its name publicy here (preventing non allowed advertising), I would like to know for who I should reach out to make this possible?

Thank you! :)

**Comment by rustdesk** [comment]
Thanks, read this https://github.com/rustdesk/rustdesk/discussions/1026  first please.

**Reply by 0x100001** [reply]
Thanks for your reply. I am also hosting all my servers with hetzner. 
Like @danielehrhardt already metioned Hetzners cloud infrastructure is perfectly fine for services like this one.
He already provides a german relay server. Do you need additional server power in germany?
If not I can still offer US-EAST (Ashburn) and Helsinki as location. 
Feel free sending me your SSH Key to nico.mak@0x101-cyber-security.de and I will invite you to the Hetzner project and deploy a server for you. There is also no limitation regarding it's uptime. :)

**Reply by rustdesk** [reply]
Noted with thanks, I will send to you.

**Reply by rustdesk** [reply]
@0x100001 I just sent you my SSH key. I prefer the US-EAST (Ashburn) server. Better if both.

**Reply by 0x100001** [reply]
@rustdesk  thank you for reaching out to me again! I invited you to the hetzner project and added your ssh key.
I did not harden the servers yet. Please tell me if you want me to do that. If you are using your own vpn server with a static IP, I can add a exception for it to the firewall to prevent ssh port scanning. I also did setup two subdomains. I can adjust the names if needed.
rustdesk-ashburn-01.0x101-cyber-security.de
rustdesk-helsinki-01.0x101-cyber-security.de

The installations are both ubuntu 22.04 and without anything done. If you need any assistance reach out for me.
Do you want to control it completely on your own? I can reguarly install updates on it if you want me to. If you like we can continue our contact via email.

**Reply by rustdesk** [reply]
Thanks, it is working now.

**Reply by 0x100001** [reply]
Fantastic! I see many are already using the servers. I will create pullrequests for the readme's in the next days. The traffic is quite high. I will keep an eye on it, maybe one day we will have to upgrade the cloud servers to dedicated servers without the 20 TB traffic limit.

**Reply by danielehrhardt** [reply]
@0x100001 i do not see a lot of traffic on my server. @rustdesk how do you route the traffic to the servers?
Ping time?

**Reply by 0x100001** [reply]
@danielehrhardt I just took a quick look with mobaxterms remote monitoring. Right now constant 30 to 40 mb down/up and around 690 active connections on the servers port. In hetzner cloud I can see it almost already consumed 1 TB in last five days.

**Reply by rustdesk** [reply]
@danielehrhardt actually your traffic is higher a bit.

**Reply by 0x100001** [reply]
@danielehrhardt Hetzner's live monitoring is sometimes not very accurate. It's something to keep an eye on before an admin in a larger company suddenly decides to move thousands of users to the home office with RustDesk and the public servers. 🤣

The current price per TB is €1.19 as far as I know, which is not much. Depending on the situation, it still makes sense to use Hetzners dedicated servers.

**Reply by danielehrhardt** [reply]
> The current price per TB is €1.19 as far as I know, which is not much. Depending on the situation, it still makes sense to use Hetzners dedicated servers.

yes, currently we have both for Rustdesk. One dedicated and one from the Hetzner Cloud.

**Reply by rustdesk** [reply]
https://twitter.com/rustdesk/status/1595597511980470272

**Reply by 0x100001** [reply]
@rustdesk 

Editing the readme's. I can remove Vultr already? What are the exact locations https://my.catixs.net/ provide? I guess Singapore?

**Comment by rustdesk** [comment]
@0x100001 lately i try to migrate our official hbbs server to Hetzner, but it does not work, because hbbs uses udp traffic. Once the udp traffic is open, the server is totally not reachable. Wondering if you know such problem.

**Reply by 0x100001** [reply]
@rustdesk never experienced that problem. Are you using hetzner cloud and maybe enabled its firewall?

**Reply by rustdesk** [reply]
azure has exactly the same issue. though the same thing works on aws and vultr.

**Reply by 0x100001** [reply]
Did you try the docker version? Maybe having it running in a container works better? Maybe the linux server itself runs into problems for some reason.

**Reply by danielehrhardt** [reply]
Have you already created a Hetzner Ticket for it?

**Reply by rustdesk** [reply]
yes, but azure support is much better.

**Reply by danielehrhardt** [reply]
Is AWS or Vultr to expensive?

**Reply by rustdesk** [reply]
@0x100001 i guess so, i will try out another os.

**Reply by rustdesk** [reply]
yes, bandwidth expensive

**Reply by rustdesk** [reply]
@0x100001 does docker bypass host tcp stack?

**Reply by danielehrhardt** [reply]
How much Bandwidth do you need?
We can also try root server if you like they have unltimited

**Reply by rustdesk** [reply]
More and more as we grow, suppose 1M/s per user, if we wanna support 10K concurrent user, it will be 10G/s.
